app/controllers/support_interface/application_forms/address_details_controller.rb

```diff
@@ -0,0 +1,38 @@
+module SupportInterface
+  module ApplicationForms
+    class AddressDetailsController < SupportInterfaceController
+      before_action :set_application_form
+
+      def edit
+        @details_form = EditAddressDetailsForm.build_from_application_form(@application_form)
+      end
+
+      def update
+        @details_form = EditAddressDetailsForm.new(
+          address_details_params.merge(address_type: @application_form.address_type),
```

Comment:
@malcolmbaig after our chat I decided to go for the approach already taken on the Candidate side.

I think this is more explicit about passing in the extra `address_type` param, and allows us to stick to the pattern of using `Form.new` in the update action.